geode-core/src/main/java/org/apache/geode/internal/cache/partitioned/RegionAdvisor.java

```diff
@@ -426,6 +430,23 @@ public void removeIdAndBuckets(InternalDistributedMember memberId, int prSerial,
     }
   }
 
+  public void removeIdAndBucket(int bucketId, InternalDistributedMember memberId, int serial,
+      boolean regionDestroyed) {
+    synchronized (preInitQueueMonitor) {
+      if (preInitQueue != null) {
+        // Queue profile during pre-initialization
+        QueuedBucketProfile qbf =
+            new QueuedBucketProfile(bucketId, memberId, serial, regionDestroyed);
+        preInitQueue.add(qbf);
+        return;
+      }
+    }
+
+    if (buckets != null) {
```

Comment:
Is it really okay to just do nothing here if buckets is null? It does not get nulled out at the end of its life. It is null after the constructor and made non-null when initializeRegionAdvisor is called. But it is called BEFORE we call processProfilesQueuedDuringInitialization which is the only place we null out preInitQueue. We only get to this null check if preInitQueue is null. My concern is if we do get to this line and buckets is null then something is wrong with our internal logic and we are not going to  call removeIdWithSerial which seems wrong.

Reply:
If `buckets` is `null`, which means `initializeRegionAdvisor()` and `processProfilesQueuedDuringInitialization()` are not called yet. At this time, as long as `RegionAdvisor` constructor is called, `preInitQueue` is not `null`. It will enqueue the bucket profile and `return` in the `if (preInitQueue != null)` block line 441.

Reply:
If we do get to this line, then `preInitQueue` has to be `null`, which means `processProfilesQueuedDuringInitialization()` is called. And `initializeRegionAdvisor()` is called before `processProfilesQueuedDuringInitialization()` to initialize the `buckets` array. So I don't think `buckets` is `null`. Maybe I am missing something?